tests/bcl-test/BCLTests/Makefile

```diff
@@ -3,18 +3,102 @@ CONFIGURATION ?= Debug
 MONO_ROOT:=../../../external/mono/
 # similar to a dict, we create diff projects, then add the assemblies we want for each of the projects
 # that way, we can match a project with more than one assembly since some test assemblies are small
-# and we reduce the compilation
-TEST_PROJECTS:=first bar baz poo
-export first_VALUE:=MONOTOUCH_System.Xml_test.dll MONOTOUCH_System.Xml.Linq_test.dll
-export bar_VALUE:=MONOTOUCH_System.Transactions_test.dll
-export baz_VALUE:=MONOTOUCH_System.Threading.Tasks.Dataflow_test.dll
-export poo_VALUE:=MONOTOUCH_System.ServiceModel_test.dll MONOTOUCH_System.ServiceModel.Web_test.dll
+# and we reduce the compilation. As it is atm we are using the same number of projects as the old
+# way to do it, so that we can get xharness working, we will modify this to improve speed
+# IMPORTANT you should notice that there are no '.' in the names, env vars do not like '.'
+TEST_PROJECTS:=System \
+			   SystemCore \
+			   SystemData \
+			   SystemNetHttp \
+			   SystemNumerics \
+			   SystemRuntimeSerialization \
+			   SystemTransactions \
+			   SystemXml \
+			   SystemXmlLinq \
+			   MonoSecurity \
+			   SystemComponentModelDataAnnotations \
+			   SystemJson \
+			   SystemServiceModelWeb \
+			   MonoDataTds \
+			   SystemIOCompression \
+			   SystemIOCompressionFileSystem \
+			   MonoCSharp \
+			   SystemSecurity \
+			   SystemServiceModel \
+
+# list of test assemblies to ignore, if the assembly is not present in one of the project or in
+# the following list, compilation will fail. We do this so that if mono adds a new test assembly
+# we do know that is being tested in the different platforms. TODO: Add an ignore per platform
+COMMON_IGNORED_TEST_ASSEMBLIES:=MONOTOUCH_Commons.Xml.Relaxng_test.dll \
+								MONOTOUCH_Cscompmgd_test.dll \
+								MONOTOUCH_I18N.CJK_test.dll \
+								MONOTOUCH_I18N.MidEast_test.dll \
+								MONOTOUCH_I18N.Other_test.dll \
+								MONOTOUCH_I18N.Rare_test.dll \
+								MONOTOUCH_I18N.West_test.dll \
+								MONOTOUCH_Mono.C5_test.dll \
+								MONOTOUCH_Mono.CodeContracts_test.dll \
+								MONOTOUCH_Mono.Parallel_test.dll \
+								MONOTOUCH_Mono.Runtime.Tests_test.dll \
+								MONOTOUCH_Mono.Tasklets_test.dll \
+								MONOTOUCH_Novell.Directory.Ldap_test.dll \
+								MONOTOUCH_System.Data.DataSetExtensions_test.dll \
+								MONOTOUCH_System.Json.Microsoft_test.dll \
+								MONOTOUCH_System.Runtime.Serialization.Formatters.Soap_test.dll \
+								MONOTOUCH_System.Threading.Tasks.Dataflow_test.dll \
+								MONOTOUCH_corlib_test.dll \
```

Comment:
why is `MONOTOUCH_corlib_test.dll` excluded ?
in fact all of the entries should have a comment / reason to be excluded